### PR TITLE
Add new topic: bundling_plugin_openapi_sources.md

### PR DESCRIPTION
**Pull request by karollewandowski**
To be merged after the feature is released.

### CODE REVIEW

File: topics/basics/bundling_plugin_openapi_sources.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+[//]: # (title: Bundling Plugin API Sources)
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+
+If a plugin exposes its own API that is meant to be used by other plugins, it is worth considering bundling the plugin sources in the ZIP distribution.
+
+If a third-party plugin uses [](tools_gradle_intellij_plugin.md) and adds a dependency to the plugin which bundles sources in the ZIP distribution, sources will be automatically attached to the plugin library and visible in the IntelliJ Platform-based IDEs when developers navigate to the API classes.
+Being able to see API sources drastically improves the development experience, and it is highly recommended to bundle them.
+
+> Attaching bundled plugin sources in IDE is available from the Gradle IntelliJ Plugin 1.7.0.
```
Comment (YannCebron) [inline topics/basics/bundling_plugin_openapi_sources.md:10]: from the -> starting with

File: topics/basics/bundling_plugin_openapi_sources.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
+[//]: # (title: Bundling Plugin API Sources)
+
+<!-- Copyright 2000-2022 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license. -->
+
+If a plugin exposes its own API that is meant to be used by other plugins, it is worth considering bundling the plugin sources in the ZIP distribution.
+
+If a third-party plugin uses [](tools_gradle_intellij_plugin.md) and adds a dependency to the plugin which bundles sources in the ZIP distribution, sources will be automatically attached to the plugin library and visible in the IntelliJ Platform-based IDEs when developers navigate to the API classes.
```
[the remark this answers is not in the document]
Comment (karollewandowski) [reply topics/basics/bundling_plugin_openapi_sources.md]: I agree and it sounds better. Fixed, thanks.

File: topics/basics/bundling_plugin_openapi_sources.md (hunk 1 above)
Comment (karollewandowski) [reply topics/basics/bundling_plugin_openapi_sources.md]: Fixed, thanks.

File: topics/basics/bundling_plugin_openapi_sources.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+Keep in mind that API should be stable and change very rarely as every incompatible change will break the client plugins.
+It is also recommended to organize the plugin code in multiple modules with clear responsibilities, e.g.:
+
+- `example-plugin-api` - a module containing API
+- `example-plugin-impl` - a module containing plugin features code that are not meant to be extended or used by client plugins
+
+General rule to define API is to include classes that are likely to be consumed by the client plugins code.
+
+Of course, more complex plugins may require more fine-grained structure.
+See [Gradle IntelliJ Plugin - Usage Examples](https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin-examples.html).
```
[the remark this answers is not in the document]
Comment (karollewandowski) [reply topics/basics/bundling_plugin_openapi_sources.md]: Fixed, thanks.

**Pull request merged**